manta-crypto/src/arkworks/serialize.rs

```diff
@@ -25,6 +28,153 @@ use {
 #[doc(inline)]
 pub use ark_serialize::*;
 
+/// Serialization Hook
+pub trait HasSerialization<'s>: 's {
+    /// Serialize Type
+    type Serialize: CanonicalSerialize + From<&'s Self>;
+}
+
+/// Deserialization Hook
+pub trait HasDeserialization: Sized {
+    /// Deserialize Type
+    type Deserialize: CanonicalDeserialize + Into<Self>;
+}
+
+/// Arkworks Reader
+pub struct ArkReader<R>
```

Comment:
What's the point of this struct? I see it's an intermediate step for `codec` implementations, but I don't understand why we need it.

Reply:
Its a way to give `codec` readers access to the `arkworks` serialization so we don't have to write the `update` function you see below every time you want to implement a serialization.